src/java/org/apache/cassandra/cql3/selection/WritetimeOrTTLSelector.java

```diff
@@ -169,28 +176,28 @@ public boolean equals(Object o)
 
         return Objects.equal(column, s.column)
             && Objects.equal(idx, s.idx)
-            && Objects.equal(isWritetime, s.isWritetime);
+            && kind == s.kind;
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hashCode(column, idx, isWritetime);
+        return Objects.hashCode(column, idx, kind);
     }
 
     @Override
     protected int serializedSize(int version)
     {
         return ByteBufferUtil.serializedSizeWithVIntLength(column.name.bytes)
                 + TypeSizes.sizeof(idx)
-                + TypeSizes.sizeof(isWritetime);
+                + TypeSizes.sizeofUnsignedVInt(kind.ordinal());
     }
 
     @Override
     protected void serialize(DataOutputPlus out, int version) throws IOException
     {
         ByteBufferUtil.writeWithVIntLength(column.name.bytes, out);
         out.writeInt(idx);
-        out.writeBoolean(isWritetime);
+        out.writeUnsignedVInt(kind.ordinal());
```

Comment:
we could make this serialization more compact. I don't think we'll have more that 256 enum types 
```suggestion
        out.writeByte(kind.ordinal());
```

Reply:
yep. In this case, `VInt` writes only 1 byte. I think it makes sense to explicitly use writeByte so to avoid checking value size.